### ISSUE TITLE
_data_dict update method

### ISSUE THREAD

**Issue by cduff4464** [Nikea/xray-vision#84]
Hey I am using some of the xray-vision library for an interface that I am making for XPD/28-ID beam line. I need the cross section 2-D view to be able to update its _data_dict so that it can take in more images while running. I am willing to make a PR for this, but I was wondering if such a method already existed in the code.

**Comment by CJ-Wright** [comment]
You might look at [this](https://github.com/Nikea/xray-vision/blob/master/xray_vision/backend/__init__.py#L215) and [this](https://github.com/Nikea/xray-vision/blob/master/xray_vision/backend/__init__.py#L252)

**Comment by cduff4464** [comment]
Tried using these, they require something called a corner list, that I am not sure how to make. I found a work around however, that lets you access the dictionary directly from where you call the cross-section 2D view class and go from there thanks though.

**Comment by CJ-Wright** [comment]
Why does it require a corner list, isn't that optional?

**Comment by cduff4464** [comment]
I tried sending in none a few times, and it refused to update. Then I tried sending in a list with some random values to no avail. Nothing made the dictionary update.

**Comment by CJ-Wright** [comment]
Hmm that's odd.
@tacaswell @ericdill Thoughts? It seems that `self.find_corners` might not exist for the [Stack classes](https://github.com/Nikea/xray-vision/blob/master/xray_vision/backend/__init__.py#L236)
Also it seems that the docstring does not match the method signature.

@cduff4464 Do you have this workaround in your branch? If not when you get a chance can you put it up? I would be very interested in looking at it.

**Comment by tacaswell** [comment]
That code looks wrong.

**Comment by cduff4464** [comment]
Yeah the workaround that I made is currently in my branch [https://github.com/cduff4464/xray-vision/blob/master/xray_vision/qt_widgets/__init__.py](url) under the new_data method in the CrossSectionMainWindow class

**Comment by CJ-Wright** [comment]
@cduff4464 Great thank you very much! Just FYI usually when you fork you create your own branch with a short name of the feature, that way the PR isn't master to master.

**Comment by cduff4464** [comment]
That is a typo on the unprotected one. I just fixed it. The self.key_list is another type, I had it set to the key_list that gets sent int that way it could be referenced in the add_data method later on. I assume it would work just fine from the messenger class. I'll make a branch and make a pull request for it.

**Comment by cduff4464** [comment]
Also really quick which messenger class did you mean in particular. I just want to make sure I add it to the right one.

**Comment by CJ-Wright** [comment]
CrossSection2DMessenger I think. Although if you find that the 1D stack also lacks an add data method you might add it to a more abstract messenger. The idea is that the issue you solved is in the messenger, as evinced by all the lines in the new method calling or setting attributes of the messenger. Thus you should make the changes in the messenger.

**Comment by cduff4464** [comment]
okay made pr with fix in it
